lecture_notes/scribe_lec2.md

```diff
@@ -0,0 +1,176 @@
+      - different nodes have different views
+      - we cannot use byzantine agreement to vote (we don't know the set of nodes)
+- **Trial 4: Bitcoin**
+  - Idea:
+    - expend some effort called proof of work
+    - make it expensive to make a block
+    - bringing incentives into the system
+    - nodes get rewarded after spending some effort to make a block
+  - (digression) Proof-of-work
+    - H(nonce|prev|transactions) < target
+    - The only way that you can find a hash that is less than that target is by doing different trials
+    - it is expensive to create two different blocks
+    - whoever finds the hash that results to be < target, gets rewarded
+  - Rewarding scheme
+    - miner includes his public key in the transaction
+    - miner is awarded new tokens + fees in the transaction
+  - Question: How does forking work?
+    - Adversary construct:
+      - Bob does a transaction that sends to Alice
+      - Bob spends the same coin to himself
+      - Bob mines two blocks
+    - In order for Bob to do this, he has to spend time and energy
+    - How to fix this?
+      - Alice only accepts the transactions only after a few blocks
+      - In order for Bob to write a parallel long chain he will have to create all the other blocks
+  - Question: When there is a fork, who choses which fork is the correct one?
+    - Always choose the longest tree
+  - Question: What does majority means?
+    - Majority of hash power! Hash/s
+    - On average someone ends up finding the hash every 10 minutes (poisson process)
+      - time to find the block = 10 mins /(my fraction of hashing power in the system)
+      - expected number blocks mined is on expectation my percentage of mining
+
+## How does Bitcoin work?
+- Limited number of tokens 21 millions
+- Supply ends in 2140
+- In 2012: 25 bitcoin rewards, 12.5 in 2016, ...
+- **Bitcoin Ecosystem**: Why don't the miners just decide to issue more Bitcoins that the protocols wants?
+  - Miners: work to produce blocks
+  - Full node: do not produce blocks but run nodes to validate blocks
+  - Users: just users!
+- **Bitcoin rules**:
+  - issuance schedule
+  - no double spending
+  - block format
+  - valid proof of work
+  - transaction format
+  - scripts
+- **Bitcoin transactions**:
+  - naive: (pk_b, sig_a(c -> pk_b))
+  - coins are consumed in their entirity (eliminated and created, never passed along)
+  - transaction have: inputs & outputs
+    - inputs: hash of previous transaction, index, script signature
+      - the input specifies the input you are consuming
+    - outputs: value, script pubkey(condition or predicate that must be satisfied) (utxos)
+      - the output is
+    - the idea is that outputs holds a predicate that must be satisfied by the input
+  - Question: How can we split coins?
+    - The transaction contains an input that "unlocks" some money in some some previous transaction
+    - The transaction also contains an output whose sum(inputs)-sum(outputs) <= 0.
```

Comment:
I think this should be "The transaction also contains outputs, and sum(inputs)-sum(outputs) must be >= 0"